transit_odp/frontend/assets/js/organisation-detail-map.js

```diff
@@ -0,0 +1,402 @@
+  "cross-icon-disruptions",
+  "diversion-icon-disruptions",
+  "engineering-icon-disruptions",
+  "event-icon-disruptions",
+  "industrial-action-icon-disruptions",
+  "question-mark-icon-disruptions",
+  "roadworks-icon-disruptions",
+  "traffic-icon-disruptions",
+  "weather-icon-disruptions",
+]
+
+const images = [
+  {url: cross, id: 'cross'},
+  {url: diversion, id: 'diversion'},
+  {url: engineering, id: 'engineering'},
+  {url: event, id: 'event'},
+  {url: industrialAction, id: 'industrial-action'},
+  {url: questionMark, id: 'question-mark'},
+  {url: roadworks, id: 'roadworks'},
+  {url: traffic, id: 'traffic'},
+  {url: weather, id: 'weather'},
+]
+
+const disruptionReasonByIcon = {
+  crossIcon: ["operatorCeasedTrading"],
+  diversionIcon: ["routeDiversion"],
+  engineeringIcon: ["emergencyEngineeringWork", "escalatorFailure", "liftFailure", "repairWork", "securityAlert", "signalFailure", "signalProblem", "vandalism"],
+  eventIcon: ["specialEvent"],
+  industrialActionIcon: ["industrialAction"],
+  questionMarkIcon: ["unknown"],
+  roadworksIcon: ["constructionWork, maintenanceWork", "roadClosed", "roadworks"],
+  trafficIcon: ["accident", "breakdown", "congestion", "incident", "overcrowded"],
+  weatherIcon: ["flooding", "fog", "heavyRain", "heavySnowFall", "highTemperatures", "ice"]
+}
+
+const disruptionReasonText = {
+  accident: "Accident",
+  securityAlert: "Security alert",
+  congestion: "Congestion",
+  roadClosed: "Road closed",
+  incident: "Incident",
+  routeDiversion: "Route diversion",
+  unknown: "Unknown",
+  vandalism: "Vandalism",
+  overcrowded: "Overcrowded",
+  operatorCeasedTrading: "Operator ceased trading",
+  roadworks: "Roadworks",
+  specialEvent: "Special event",
+  industrialAction: "Industrial action",
+  signalProblem: "Signal problem",
+  signalFailure: "Signal failure",
+  repairWork: "Repair work",
+  constructionWork: "Construction work",
+  maintenanceWork: "Maintenance work",
+  emergencyEngineeringWork: "Emergency engineering work",
+  escalatorFailure: "Escalator failure",
+  liftFailure: "Lift failure",
+  fog: "Fog",
+  heavySnowFall: "Heavy snowfall",
+  heavyRain: "Heavy rain",
+  ice: "Ice",
+  highTemperatures: "High temperatures",
+  flooding: "Flooding",
+}
+
+const httpGetAsync = (theUrl, callback) => {
+  const request = new XMLHttpRequest();
+
+  if (!request) {
+    alert("Giving up :( Cannot create an XMLHTTP instance");
+    return false;
+  }
+  request.onreadystatechange = function () {
+    if (request.readyState === 4 && request.status === 200)
+      callback(request.responseText);
+  };
+  request.open("GET", theUrl, true); // true for asynchronous
+  request.send();
+};
+
+const initOrgMap = (apiRoot, orgId) => {
+  const servicePatternUrl =
+    apiRoot + "organisation_map_data/?pk=" + orgId.toString();
+
+  // Initialise Map
+  mapboxgl.accessToken =
+    "pk.eyJ1IjoiaGFsYmVydHJhbSIsImEiOiJjaXFiNXVnazIwMDA0aTJuaGxlaTU1M2ZtIn0.85dXvyj6V2LbBFvXfpQyYA";
```

Comment:
I've hardcoded it as this is how it was being done for the other maps within BODS. I can look into passing this through if needed.

Reply:
As discussed on teams, there is a ticket outstanding to update how the mapbox API key is handled across the BODS site: https://kpmgengineering.atlassian.net/browse/BODS-831

I would prefer to leave it hardcoded for now, until all maps can be updated. I have added a comment to ticket BODS-831 so that it is clear that the disruptions maps will need updating also.